src/main/java/org/apache/sling/xss/impl/xml/MapBuilder.java

```diff
@@ -0,0 +1,247 @@
+            "script",
+            "object",
+            "applet",
+            "frame",
+            "base",
+            "param",
+            "meta",
+            "input",
+            "textarea",
+            "embed",
+            "basefont",
+            "col");
+
+    public void createRulesMap(PolicyProvider policy, AntiSamyRules topLevelElement) throws PolicyException {
+        this.policy = policy;
+
+        parseCommonRegExps(topLevelElement.getRegexpList());
+        parseDirectives(topLevelElement.getDirectiveList());
+        parseAllowedEmptyTags(topLevelElement.getAllowedEmptyTags());
+        parseCommonAttributes(topLevelElement.getCommonAttributeList());
+        parseGlobalAttributes(topLevelElement.getGlobalTagAttributes().getGlobalTagAttributeList());
+        parseDynamicAttributes(topLevelElement.getDynamicTagAttribute().getDynamicTagAttributeList());
+        parseTagRules(topLevelElement.getTagRulesList());
+
+        parseCSSRules(topLevelElement.getPropertyList());
+    }
+
+    /**
+     * Go through the <common-regexps> section of the policy file.
+     *
+     * @param root                      Top level of <common-regexps>
+     */
+    private void parseCommonRegExps(List<Regexp> root) {
+        for (Regexp regex : root) {
+            String name = regex.getName();
+            Pattern regexp = Pattern.compile(regex.getValue(),
+                    Pattern.DOTALL);
+            policy.commonRegularExpressions.put(name, regexp);
+        }
+    }
+
+    /**
+     * Go through <directives> section of the policy file.
+     *
+     * @param root       Top level of <directives>
+     */
+    private void parseDirectives(List<Directive> root) {
+        for (Directive directive : root) {
+            String name = directive.getName();
+            String value = directive.getValue();
+            policy.directives.put(name, value);
+        }
+    }
+
+    private void parseCommonAttributes(List<Attribute> root) {
+        for (Attribute attribute : root) {
+            List<Regexp> allowedRegexps = getAllowedRegexps(attribute.getRegexpList());
+            Attribute newAttribute = new Attribute(attribute.getName(), allowedRegexps, attribute.getLiteralList(),
+                    attribute.getOnInvalid(), attribute.getDescription());
+            policy.commonAttributes.put(attribute.getName().toLowerCase(), newAttribute);
+        }
+    }
+
+    // /**
+    // * Go through <allowed-empty-tags> section of the policy file.
+    // *
+    // * @param allowedEmptyTagsListNode Top level of <allowed-empty-tags>
+    // * @param allowedEmptyTags The tags that can be empty
+    // */
+    private void parseAllowedEmptyTags(AllowedEmptyTags allowedEmptyTagsList) throws PolicyException {
+        if (allowedEmptyTagsList != null) {
+            policy.allowedEmptyTags = allowedEmptyTagsList.getLiterals();
+        } else
+            policy.allowedEmptyTags.addAll(ALLOWED_EMPTY_TAGS);
+    }
+
+    // /**
```

Comment:
Remove the leading '//' so this can be recognised as a javadoc block.

Reply:
Resolved: 99bb91ab6e2529fd6ef25858f874a26cb159e87e